hooks/tk-multi-publish2/basic/nuke_publish_reviewsubmission.py

```diff
@@ -0,0 +1,333 @@
+
+        <ul>
+        <li><code>filename.v###.ext</code></li>
+        <li><code>filename_v###.ext</code></li>
+        <li><code>filename-v###.ext</code></li>
+        </ul>
+
+        After publishing, if a version number is detected in the work file, the
+        work file will automatically be saved to the next incremental version
+        number. For example, <code>filename.v001.ext</code> will be published
+        and copied to <code>filename.v002.ext</code>
+
+        If the next incremental version of the file already exists on disk, the
+        validation step will produce a warning, and a button will be provided in
+        the logging output which will allow saving the session to the next
+        available version number prior to publishing.
+
+        <br><br><i>NOTE: any amount of version number padding is supported. for
+        non-template based workflows.</i>
+
+        <h3>Overwriting an existing publish</h3>
+        In non-template workflows, a file can be published multiple times,
+        however only the most recent publish will be available to other users.
+        Warnings will be provided during validation if there are previous
+        publishes.
+        """ % (loader_url,)
+
+    @property
+    def settings(self):
+        """
+        Dictionary defining the settings that this plugin expects to receive
+        through the settings parameter in the accept, validate, publish and
+        finalize methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
+        part of its environment configuration.
+        """
+
+        # inherit the settings from the base publish plugin
+        base_settings = super(NukeReviewSubmissionPublishPlugin, self).settings or {}
+
+        # settings specific to this class
+        nuke_publish_settings = {
+            "Publish Template": {
+                "type": "template",
+                "default": None,
+                "description": "Template path for published work files. Should"
+                               "correspond to a template defined in "
+                               "templates.yml.",
+            }
+        }
+
+        # update the base settings
+        base_settings.update(nuke_publish_settings)
+
+        return base_settings
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["maya.*", "file.maya"]
+        """
+        return ["nuke.reviewsubmission"]
```

Comment:
Would be nice to attach this to `*.sequence` items

Reply:
changed